lib/PoolClusterFarm.js

```diff
@@ -0,0 +1,120 @@
+var Pool          = require('./Pool');
+var PoolConfig    = require('./PoolConfig');
+var PoolNamespace = require('./PoolNamespace');
+var PoolSelector  = require('./PoolSelector');
+var Util          = require('util');
+var EventEmitter  = require('events').EventEmitter;
+
+/**
+ * PoolClusterFarm
+ * @constructor
+ * @param {object} [config] The pool cluster farm configuration
+ * @public
+ */
+function PoolClusterFarm(config) {
+  EventEmitter.call(this);
+
+  config = config || {};
+  this._servers = [];
+  this._numMasters = 0;
+  this._numSlaves = 0;
+  this._last_index = 0;
+  this.debug = config.debug || false
+}
+
+Util.inherits(PoolClusterFarm,EventEmitter);
+
+PoolClusterFarm.prototype.query = function query(sql,values,cb) {
+	var self = this
+
+	if (!this._numMasters || this._numMasters < 1) {
+		if (cb && typeof cb == 'function') { return cb('No servers assigned to PoolClusterFarm. Please verify...'); }
+		else { return values('No servers assigned to PoolClusterFarm. Please verify...'); }
+	}
+
+	if (typeof sql != 'string' || sql.length < 7) {
+		if (cb && typeof cb == 'function') { return cb('Invalid sql statement. Please verify...'); }
+		else { return values('Invalid sql statement. Please verify...'); }
+	}
+
+	if (sql.substr(0,6).toLowerCase() == 'select') {
+		var index_to_use = this._last_index;
+		if (index_to_use >= (this._servers.length)) { this._last_index = index_to_use = 0; }
+		else { this._last_index++; }
+		if (this.debug) { console.log('Using server',(index_to_use+1),'of',(this._servers.length)); }
+		this._servers[index_to_use].pool.query(sql,values,cb);
+	} else {
+		var num = this._numMasters;
+
+		var updateMasters = function(i) {
+			self._servers[i].pool.query(sql,values,function(err,rows) {
+				if (!(--num)) {
+					if (cb && typeof cb == 'function') { return cb(err,rows); }
```

Comment:
ok I monitor errors and immediately returns with the error and stops all other execution. I'm not certain this is the best solution but it does accomplish the goal.